### PR TITLE
Gizmo transform shape toggling

### PR DESCRIPTION
**Pull request by kpal81xd**
Allows the transform shapes in the gizmo to be enabled or disabled using `enableShape` and `isShapeEnabled` methods.

Example (Disabled all lines and planes except the center)
image

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: extras/gizmo/axis-shapes.js [hunk 1]
```diff
@@ -106,10 +110,26 @@ class AxisShape {
...
         this._defaultColor = options.defaultColor ?? Color.BLACK;
         this._hoverColor = options.hoverColor ?? Color.WHITE;
+        this._disabledColor = options.disabledColor ?? Color.GRAY_COLOR;
+    }
+
+    set disabled(value) {
+        this._disabled = value ?? false;
+        if (value) {
+            for (let i = 0; i < this.meshInstances.length; i++) {
+                this.meshInstances[i].material = this._disabledColor;
```
Comment (slimbuck) [inline extras/gizmo/axis-shapes.js:126]: Do you need to update material going from enabled to disabled as well (currently that's skipped)?

Comment (slimbuck) [reply extras/gizmo/axis-shapes.js:126]: Also a little confusing that you're setting material to a color :)

Comment (kpal81xd) [reply extras/gizmo/axis-shapes.js]: Oh yea true it will not be updated until hover is called 👍

File: extras/gizmo/axis-shapes.js [hunk 2]
```diff
@@ -106,10 +110,24 @@ class AxisShape {
...
 
+        this._disabled = options.disabled ?? false;
+
         this._layers = options.layers ?? this._layers;
 
-        this._defaultColor = options.defaultColor ?? Color.BLACK;
-        this._hoverColor = options.hoverColor ?? Color.WHITE;
+        this._defaultMaterial = options.defaultMaterial ?? Color.BLACK;
+        this._hoverMaterial = options.hoverMaterial ?? Color.WHITE;
+        this._disabledMaterial = options.disabledMaterial ?? Color.GRAY_COLOR;
```
Comment (willeastcott) [inline extras/gizmo/axis-shapes.js:119]: Shouldn't this be `GRAY`?

File: extras/gizmo/gizmo.js [hunk 3]
```diff
@@ -25,14 +25,14 @@ const ORTHO_SCALE_RATIO = 0.32;
...
-export const LOCAL_COORD_SPACE = 'local';
+export const GIZMO_LOCAL = 'local';
 
 /**
  * World coordinate space.
  *
  * @type {string}
  */
-export const WORLD_COORD_SPACE = 'world';
+export const GIZMO_GLOBAL = 'world';
```
Comment (willeastcott) [inline extras/gizmo/gizmo.js:35]: `GIZMO_GLOBAL` -> `GIZMO_WORLD`.

File: extras/gizmo/axis-shapes.js [hunk 4]
```diff
@@ -106,10 +110,24 @@ class AxisShape {
         this._rotation = options.rotation ?? new Vec3();
         this._scale = options.scale ?? new Vec3(1, 1, 1);
 
+        this._disabled = options.disabled ?? false;
+
         this._layers = options.layers ?? this._layers;
 
-        this._defaultColor = options.defaultColor ?? Color.BLACK;
-        this._hoverColor = options.hoverColor ?? Color.WHITE;
+        this._defaultMaterial = options.defaultMaterial ?? Color.BLACK;
```
Comment (mvaligursky) [inline extras/gizmo/axis-shapes.js:117]: so is this a material or a color?

File: extras/gizmo/axis-shapes.js [hunk 5]
```diff
@@ -106,10 +110,24 @@ class AxisShape {
...
+        this._hoverMaterial = options.hoverMaterial ?? Color.WHITE;
+        this._disabledMaterial = options.disabledMaterial ?? Color.GRAY;
+    }
+
+    set disabled(value) {
+        for (let i = 0; i < this.meshInstances.length; i++) {
+            this.meshInstances[i].material = value ? this._disabledMaterial : this._defaultMaterial;
```
Comment (mvaligursky) [inline extras/gizmo/axis-shapes.js:124]: so here you assign `Color.Black` as a material to mesh instance?

File: extras/gizmo/gizmo.js [hunk 6]
```diff
@@ -165,12 +165,17 @@ class Gizmo extends EventHandler {
     _scale = 1;
 
     /**
-     * Internal version of coordinate space. Defaults to {@link WORLD_COORD_SPACE}.
+     * Internal version of coordinate space. Can be:
+     *
+     * {@link GIZMO_LOCAL}
```
Comment (mvaligursky) [inline extras/gizmo/gizmo.js:170]: the `Can be` part needs to be on the public setter, and not on a private property.

Comment (mvaligursky) [reply extras/gizmo/gizmo.js:170]: including the default and all that. Here just a type is needed.

File: extras/gizmo/axis-shapes.js (hunk 5 above)
Comment (kpal81xd) [reply extras/gizmo/axis-shapes.js:135]: Misnaming error 👍

**Review by willeastcott — APPROVED** [review]

**Review by mvaligursky — APPROVED** [review]

**Pull request merged**